solr/core/src/java/org/apache/solr/handler/component/QueryComponent.java

```diff
@@ -401,6 +403,14 @@ public void process(ResponseBuilder rb) throws IOException
     doProcessUngroupedSearch(rb, cmd, result);
   }
 
+  private int getMinExactHits(SolrParams params) {
+    long minExactHits = params.getLong(CommonParams.MIN_EXACT_HITS, Integer.MAX_VALUE);
+    if (minExactHits < 0 || minExactHits > Integer.MAX_VALUE) {
```

Comment:
This is intentional. See the discussion with Adrien in this same PR. We allow longs just because the `minExactHits` is in relation to the `numFound`, which is a long, however, any value greater than `Integer.MAX_VALUE` doesn't make sense, since Lucene doesn't allow more than that in a single shard. In case of distributed queries, the `minExactHits` is used by every shard